tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/TeamMailboxMemberSetRequest.scala

```diff
@@ -0,0 +1,100 @@
+package com.linagora.tmail.james.jmap.model
+
+import cats.implicits._
+import com.linagora.tmail.team.{TeamMailbox, TeamMailboxMember}
+import org.apache.james.core.Username
+import org.apache.james.jmap.core.SetError.SetErrorDescription
+import org.apache.james.jmap.core.{AccountId, SetError}
+import org.apache.james.jmap.method.WithAccountId
+
+import scala.util.Try
+
+trait ParsingRequestException extends RuntimeException {
+  def message: String
+
+  override def getMessage: String = message
+}
+case class InvalidTeamMailboxException(teamMailboxNameDTO: TeamMailboxNameDTO) extends ParsingRequestException {
+  override def message: String = s"Invalid teamMailboxName"
+}
+
+case class InvalidRoleException(role: TeamMailboxMemberRoleDTO) extends ParsingRequestException {
+  override def message: String = s"Invalid role: ${role.role}"
+}
+
+case class InvalidTeamMemberNameException(memberName: TeamMailboxMemberName) extends ParsingRequestException {
+  override def message: String = s"Invalid team member name: ${memberName.value}"
+}
+
+trait ParsingRequestResult
```

Comment:
Should better if has prefix name
eg: `TeamMailboxMemberParsingRequestResult`

or relocate it to child package
eg: `com.linagora.tmail.james.jmap.model.teammailbox.member.ParsingRequestResult`